bitnami/apache/templates/hpa.yaml

```
@@ -0,0 +1,34 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: {{ .Values.autoscaling.apiVersion }}
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
```

Do no hardcode the apiVersion, instead use our "common.capabilities.deployment.apiVersion" macro.
```suggestion
    apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
```

---

:sweat_smile: this was my first idea, but then I noticed the `apiVersion` was hardcoded [in the Deployment](https://github.com/bitnami/charts/blob/85eeac7ca20ec1dfc5330b70babac15a81c79003/bitnami/apache/templates/deployment.yaml#L1) we're targeting. I am concerned using two different mechanism to set that value might lead to edge cases where the HPA doesn't target the deployment at all.

What do you think? Would you prefer to keep it hardcoded everywhere, or should I change the deployment to use the `include` logic as well?

---

Yes, updating the deployment is the right approach.